cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Certificates.java

```diff
@@ -0,0 +1,274 @@
+            sbj.setOrganizationName("io.strimzi");
+            sbj.setCommonName(name);
+
+            sbj.setSubjectAltNames(sanFn.apply(i));
+
+            certManager.generateCsr(brokerKeyFile, brokerCsrFile, sbj);
+            certManager.generateCert(brokerCsrFile, clusterCa.key(), clusterCa.cert(), brokerCertFile,
+                    sbj, ModelUtils.getCertificateValidity(kafka));
+
+            certs.put(podName,
+                    new CertAndKey(Files.readAllBytes(brokerKeyFile.toPath()), Files.readAllBytes(brokerCertFile.toPath())));
+        }
+
+        if (!brokerCsrFile.delete()) {
+            log.warn("{} cannot be deleted", brokerCsrFile.getName());
+        }
+        if (!brokerKeyFile.delete()) {
+            log.warn("{} cannot be deleted", brokerKeyFile.getName());
+        }
+        if (!brokerCertFile.delete()) {
+            log.warn("{} cannot be deleted", brokerCertFile.getName());
+        }
+
+        return certs;
+    }
+
+    public Map<String, CertAndKey> generateZkCerts(Kafka kafka) throws IOException {
+        String cluster = kafka.getMetadata().getName();
+        String namespace = kafka.getMetadata().getNamespace();
+        Function<Integer, Map<String, String>> san = i -> {
+            Map<String, String> sbjAltNames = new HashMap<>();
+            sbjAltNames.put("DNS.1", ZookeeperCluster.serviceName(cluster));
+            sbjAltNames.put("DNS.2", String.format("%s.%s.svc.%s", ZookeeperCluster.serviceName(cluster), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
+            sbjAltNames.put("DNS.3", String.format("%s.%s.%s.svc.%s", ZookeeperCluster.zookeeperPodName(cluster, i),
+                    ZookeeperCluster.headlessServiceName(cluster), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
+            return sbjAltNames;
+        };
+
+        log.debug("Cluster communication certificates");
+        return maybeCopyOrGenerateCerts(
+            kafka.getSpec().getZookeeper().getReplicas(),
+            ZookeeperCluster.zookeeperClusterName(cluster),
+            san,
+            kafka,
+            zkNodesSecret,
+            podNum -> ZookeeperCluster.zookeeperPodName(cluster, podNum));
+    }
+
+    public Map<String, CertAndKey> generateBrokerCerts(Kafka kafka) throws IOException {
+        String cluster = kafka.getMetadata().getName();
+        String namespace = kafka.getMetadata().getNamespace();
+        log.debug("Internal communication certificates");
+        Function<Integer, Map<String, String>> san = i -> {
+            Map<String, String> sbjAltNames = new HashMap<>();
+            sbjAltNames.put("DNS.1", KafkaCluster.serviceName(cluster));
+            sbjAltNames.put("DNS.2", String.format("%s.%s.svc.%s", KafkaCluster.serviceName(cluster), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
+            sbjAltNames.put("DNS.3", String.format("%s.%s.%s.svc.%s", KafkaCluster.kafkaPodName(cluster, i), KafkaCluster.headlessServiceName(cluster), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
```

Comment:
Pay attention here because @scholzj added more entries in the PR about exposing Kafka.